setup.py

```diff
@@ -1,73 +1,102 @@
 #!/usr/bin/env python3
+import io
 import os.path
+import re
 import stat
 import sys
+import tarfile
+import tempfile
+import zipfile
 from distutils.command.build import build as orig_build
 from distutils.core import Command
-from hashlib import sha512
 from http import HTTPStatus
 from urllib.request import urlopen
 
 from setuptools import setup
 from setuptools.command.install import install as orig_install
 
-SHELLCHECK_VERSION = '0.7.0'
+SHELLCHECK_VERSION = '0.7.1'
 PY_VERSION = '1'
 
 
 def get_arch() -> str:
     if sys.platform.startswith('linux'):
-        # TODO(rhee): detect "linux-aarch64" and "linux-armv6hf"
-        return 'linux-x86_64'
+        # TODO(rhee): detect "linux.aarch64" and "linux.armv6hf"
+        return 'linux.x86_64.tar.xz'
     elif sys.platform == 'darwin':
-        return 'darwin-x86_64'
+        return 'darwin.x86_64.tar.xz'
     elif sys.platform == 'win32':
-        return 'exe'
+        return 'zip'
     else:
         raise ValueError('Unsupported platform')
 
 
-def get_executable_name() -> str:
+def get_artifact_name() -> str:
     return f'shellcheck-v{SHELLCHECK_VERSION}.{get_arch()}'
 
 
 def get_download_url() -> str:
-    return f'https://storage.googleapis.com/shellcheck/{get_executable_name()}'
+    return (
+        'https://github.com/koalaman/shellcheck/releases/download/'
+        f'v{SHELLCHECK_VERSION}/{get_artifact_name()}'
+    )
 
 
 def download(url: str) -> bytes:
-    with urlopen(url) as resp, urlopen(url + '.sha512sum') as shasum_resp:
+    with urlopen(url) as resp:
         code = resp.getcode()
-        scode = shasum_resp.getcode()
-        if code != HTTPStatus.OK or scode != HTTPStatus.OK:
-            raise ValueError(f'HTTP failure. Codes: {code}, {scode}')
-
-        data = resp.read()
-        actual_shasum = sha512(data).hexdigest()
-        expected_shasum = shasum_resp.read().decode('utf8').split(' ')[0]
-
-        if expected_shasum != actual_shasum:
-            raise ValueError(
-                f'SHA512 mismatch! '
-                f'expected "{expected_shasum}" but got "{actual_shasum}"',
-            )
-
-        return data
+        if code != HTTPStatus.OK:
+            raise ValueError(f'HTTP failure. Code: {code}')
+        return resp.read()
+
+
+def extract(url: str, data: bytes):
+    result = {}
+    if re.search(r'(?i)\.tar\.', url):
+        tmpf = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
this uses delete=False and then always deletes it -- should just use the normal context manager

Reply:
If we want to change to in-memory tarfile, this can go away entirely.

Reply:
If we specify delete=False, context manager won't delete the file on close. And I wanted it closed after writing (the inner "with tmpf" block does that), because NamedTemporaryFile docs say it cannot be opened another time on Windows while it's still open. And as long as tarfile operates on file names, we need the file to exist after it is written to and closed. Leaving as is because with those constraints I don't see how to simplify this considerably.